documentation/developer_setup.md

```diff
@@ -0,0 +1,336 @@
+
+The recording files eventually live in the main repository (`github.com/Azure/azure-sdk-for-go`) and we need to make sure that all of these recordings are free from secrets. To do this we use Sanitizers with regular expressions for replacements. All of the available sanitizers are available as methods from the `recording` package. The recording methods generally take three parameters: the test instance (`t *testing.T`), the value to be removed (ie. an account name or key), and the value to use in replacement.
+
+| Sanitizer Type | Method | Parameters | Notes |
+| -------------- | ------ | ---------- | ----- |
+<!-- | BodyKeySanitizer | `recording.AddBodyKeySanitizer(t, ...)` | ... | ... |
+| BodyRegexSanitizer | `recording.BodyRegexSanitizer(t, ...)` | ... | ... |
+| ContinuationSanitizer | `recording.ContinuationSanitizer(t, ...)` | ... | ... |
+| GeneralRegexSanitizer | `recording.GeneralRegexSanitizer(t, ...)` | ... | ... |
+| HeaderRegexSanitizer | `recording.HeaderRegexSanitizer(t, ...)` | ... | ... |
+| OAuthResponseSanitizer | `recording.OAuthResponseSanitizer(t, ...)` | ... | ... |
+| RemoveHeaderSanitizer | `recording.RemoveHeaderSanitizer(t, ...)` | ... | ... |
+| ReplaceRequestSubscriptionId | `recording.ReplaceRequestSubscriptionId(t, ...)` | ... | ... | -->
+| UriRegexSanitizer | `recording.UriRegexSanitizer(t, ...)` | ... | ... |
+
+Note that removing the names of accounts and other values in your recording can have side effects when running your tests in playback. To take care of this, there are additional methods in the `internal/recording` module for reading environment variables and defaulting to the processed recording value. For example, if the `aztable` library had a test for creating a client and "requiring" the account name to be the same as provided it could potentially look similar to this:
+
+```golang
+func TestClient(t *testing.T) {
+	accountName := recording.GetEnvVariable(t, "TABLES_PRIMARY_ACCOUNT_NAME", "fakeAccountName")
+	// If running in playback, the value is "fakeAccountName". If running in "record" the value is whatever is stored in the environment variable
+	accountKey := recording.GetEnvVariable(t, "TABLES_PRIMARY_ACCOUNT_KEY", "fakeAccountKey")
+	cred, err := NewSharedKeyCredential(accountName, accountKey)
+	require.NoError(t, err)
+
+	client, err := NewClient("someTableName", someServiceURL, cred, nil)
+	require.NoError(t, err)
+	require.Equal(t, accountName, client.AccountName())
+}
+```
+
+### Using `azidentity` Credentials In Tests
+
+The credentials in `azidentity` are not automatically configured to run in playback mode. To make sure your tests run in playback mode even with `azidentity` credentials the best practice is to use a simple `FakeCredential` type that inserts a fake Authorization header to mock a credential. An example for swapping the `DefaultAzureCredential` using a helper function is shown below in the context of `aztable`
+
+```golang
+type FakeCredential struct {
```

Comment:
Not directly related to this PR, but this makes me think there's a need to have a `mock.TokenCredential` in the `internal` module.  Would existing/future tests benefit from this?

Reply:
Yes, but would we be able to do this without adding a circular dependency between azcore and internal?

Reply:
Oh right because of the `runtime.AuthenticationOptions` param.  This might end up being resolved with the pending auth changes @chlowell is working on.